benchmark_runner/common/clouds/BM/bare_metal_operations.py

```diff
@@ -0,0 +1,284 @@
+        :return:
+        """
+        self._remote_ssh.run_command('podman pod restart pod_ci')
+
+    @typechecked()
+    @logger_time_stamp
+    def update_ocp_version(self, file_name: str):
+        """
+        This method update the ocp version on provision machine
+        :param file_name:
+        :return:
+        """
+        # Get the latest assisted installer version
+        if self.LATEST in self._install_ocp_version:
+            self._install_ocp_version = self._get_installation_version()
+        openshift_version_data = self._install_ocp_version.split('.')
+        self._remote_ssh.replace_parameter(remote_path='/root/jetlag/ansible/vars',
+                                            file_name=file_name,
+                                            parameter='ocp_release_image:',
+                                            value=f'quay.io\/openshift-release-dev\/ocp-release:{self._install_ocp_version}-x86_64')
+        self._remote_ssh.replace_parameter(remote_path='/root/jetlag/ansible/vars',
+                                            file_name=file_name,
+                                            parameter='openshift_version:',
+                                            value=f'"{openshift_version_data[0]}.{openshift_version_data[1]}"')
+
+    def update_provision_config(self):
+        """
+        This method update provision config file
+        :return:
+        """
+        # update ssh config - must add it and also mount it
+        self._ssh.run(f"echo Host provision > /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'HostName {self._provision_ip} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'User {self._user} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'IdentityFile {self._container_private_key_path} >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'StrictHostKeyChecking no >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'ServerAliveInterval 30 >> /{self._user}/.ssh/config")
+        self._ssh.run(f"echo -e '\t'ServerAliveCountMax 5 >> /{self._user}/.ssh/config")
+        self._ssh.run(f"chmod 600 /{self._user}/.ssh/config")
+
+    @logger_time_stamp
+    def run_ocp_installer(self):
+        """
+        This method run ocp assisted installer
+        :return: True if installation success and raise exception if installation failed
+        """
+        """
+        This method run ocp assisted installer
+        :return: True if installation success and raise exception if installation failed
+        """
+        self.update_provision_config()
+        logger.info(f'Starting OCP assisted installer, Start time: {datetime.now().strftime(datetime_format)}')
+        # Must add -t otherwise remote ssh of ansible will not end
+        self._ssh.run(f"ssh -t provision \"{self._install_ocp_cmd()}\" ")
```

Comment:
Does this raise an exception if it fails?  If so, you should catch the exception here.  You can re-raise it, but if the ocp installer fails, it's cleaner to catch it here.